com.unity.netcode.gameobjects/Runtime/NetworkVariable/ClientNetworkVariable.cs

```diff
@@ -40,7 +40,7 @@ public override T Value
 
                 // Also, note this is not really very water-tight, if you are running as a host
                 //  we cannot tell if a ClientNetworkVariable write is happening inside server-ish code
-                if (m_NetworkBehaviour.NetworkManager.IsServer)
+                if (m_NetworkBehaviour.NetworkManager.IsServer && !m_NetworkBehaviour.NetworkManager.IsHost)
```

Comment:
This check could be simplified to `!IsClient`.

Reply:
haha, +1 :)